iib/workers/s3_utils.py

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import logging
+
+import boto3
+
+from iib.exceptions import IIBError
+from iib.workers.config import get_worker_config
+
+log = logging.getLogger(__name__)
+
+conf = get_worker_config()
+
+
+def upload_file_to_s3_bucket(file_path, key_prefix, file_key):
+    """
+    Upload artifact file to AWS S3 bucket.
+
+    :param str file_path: the path of the file locally where the artifact file is stored
+    :param str key_prefix: the logical location of the file in the S3 bucket
+    :param str key_prefix: the name of the file in S3 bucket
+    :raises IIBError: when unable to upload file to the S3 bucket
+    """
+    try:
+        log.info(
+            'Uploading file %s/%s to S3 bucket: %s',
+            key_prefix,
+            file_key,
+            conf['iib_aws_s3_bucket_name'],
```

Comment:
Here you are uploading whatever `{key_prefix}/{file_key}` to S3 which can lead to any function creating structures as they want and eventually causing mess on S3 tree.

Wouldn't it be better to limit some specific `key_prefix` which the function accepts? Then you prevent mess on S3.

Let me know what do you think about it.

Reply:
So S3 doesn't have a tree. So when we create something like `{key_prefix}/{file_key}` it's technically a file with name `{key_prefix}/{file_key}`. So I think it wouldn't matter if we limit some specific values or not. That's my two cents. please let me know if you feel strongly against it.

Reply:
No problem at all.

S3 uses APIs to manage the FS but if you look at AWS Console you'll see that when sending a file with a `{key_prefix}` before it S3 kinda "creates" a directory, it's not exactly a directory, but look likes on console.

Anyway it's not a huge problem for me, you can leave it as it is.

Reply:
Any mess that we create on s3 will be self-inflicted since the `key_prefix` and `file_key` are not user provided parameters. I don't think restricting to certain values will significantly reduce the chances of us making a mess.